enacts/onset/agronomy.py

```diff
@@ -128,43 +135,79 @@ def soil_plant_water_balance(
         * station        (station) int64 0 1
     """
     
-    # First Step
-    if np.size(et) == 1:
-        et = xr.DataArray(et)
-    # Setting Kc
+    # Initializations
+    et = xr.DataArray(et)
+    taw = xr.DataArray(taw)
+    sminit = xr.DataArray(sminit)
     if kc_params is None:
+        if planting_date is not None or sm_threshold is not None:
+            raise Exception(
+                "if Kc is not defined, neither planting_date nor sm_threshold should be"
+            )
         kc = 1
-    else:
+        et_crop = et
+    else: #et_crop depends on et, kc and planting_date dims, and time_dim
         kc_inflex = kc_params.assign_coords(
             kc_periods=kc_params["kc_periods"].cumsum(dim="kc_periods")
         )
-        if planting_date is not None:
+        if planting_date is not None: # distance between 1st and planting days
+            if sm_threshold is not None:
+                raise Exception("either planting_date or sm_threshold should be defined")
             planted_since = peffective[time_dim][0].drop_vars(time_dim) - planting_date
-            kc = kc_inflex.interp(
-                kc_periods=planted_since, kwargs={"fill_value": 1}
-            ).drop_vars("kc_periods")
-    # Initializations of sm, drainage and et_crop
-    et_crop0 = kc * et.isel({time_dim: 0}, missing_dims='ignore', drop=True)
-    sm0, drainage0 = soil_plant_water_step(
-        sminit,
-        peffective.isel({time_dim: 0}, drop=True),
-        et_crop0,
-        taw,
+        else: # 1st day is planting day if sminit met condition
+            if sm_threshold is not None:
+                planted_since = xr.where(
+                    sminit >= sm_threshold, 0, np.nan
+                ).astype("timedelta64[D]")
+            else:
+                raise Exception("if planting_date is not defined, then define a sm_threshold")
+        et_crop, trash1, trash2, trash3 = xr.broadcast(
```

Comment:
I saw the broadcast_like function and indeed figured I was going to have to call it several times and thought maybe it was not meant for that usage... so I used broadcast (for which I thought there was a way to ignore the other outputs but it appeared not after all).

Your favorite solution is almost an independent problem. You are only saying not to broadcast the time dimension. We still need to do all the broadcasting gymnastics for all the other dimensions to initialize

At some point I wanted to run the water balance globally :)
But you know, Ethiopia ENACST is like 400 by 300 so it's bigger than a 1 degree global dataset...

Maybe I do broadcast_like and we move on? This being said, we are now using this for one year only, since we will want to use if for say 40 years... so it may be worth asking the right optimization questions now.

Reply:
I switched to broadcast_like.